repoze/xmliter/serializer.py

```diff
@@ -21,7 +21,10 @@ def __init__(self, tree, serializer=None, pretty_print=False, doctype=None):
     def serialize(self, encoding=None):
         # Defer to the xsl:output settings if appropriate
         if isinstance(self.tree, lxml.etree._XSLTResultTree):
-            result = str(self.tree)
+            if encoding is str:
+                result = str(self.tree)
+            else:
+                result = bytes(self.tree)
```

Comment:
I've got no real problem with this change, but it feels like it should require one mor more new tests:  either that, or we have really poor assertions going, such that the change to use `bytes` by default is getting papered over.

Reply:
I already added a [test in the first commit](https://github.com/repoze/repoze.xmliter/pull/18/commits/f9ffff7a479119bb4308a0f7062c31285e752b31#diff-084016a78445faf2d29b62ca569ebcaf95693c8f5127913c443d677cdb19cb01R159).
This extra line is not very noticeable when you look at the diff of the full PR, because the second commit makes more Python 3 changes in the same test file.

Doing the test on master fails:

```
    def test_xsl(self):
        t = utils.getHTMLSerializer(self.create_iterable(body='<br>'))
        transform = lxml.etree.XSLT(lxml.etree.XML(b'''
            <xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
                <xsl:output method="xml" indent="no" omit-xml-declaration="yes"
                    media-type="text/html" encoding="UTF-8"
                    doctype-public="-//W3C//DTD XHTML 1.0 Transitional//EN"
                    doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"
                    />
                <xsl:template match="@*|node()">
                  <xsl:copy>
                    <xsl:apply-templates select="@*|node()"/>
                  </xsl:copy>
                </xsl:template>
            </xsl:stylesheet>
            '''))
        t.tree = transform(t.tree)
        self.assertTrue('<br />' in str(t))
>       self.assertTrue(b'<br />' in bytes(t))
E       TypeError: __bytes__ returned non-bytes (type str)
```

This error already happens when simply calling `bytes(t)`.

Reply:
OK, I see.  I missed the actual new assertion in the middle of that sea of janitorial changes in the second commit.